sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/DeadStores.cs

```diff
@@ -416,6 +416,65 @@ public virtual int Create()
         }
     }
 
+    public class AkkaSnippet
+    {
+        internal void OnlyFinally(object actor)
+        {
+            try
+            {
+                Foo(actor);
+            }
+            finally
+            {
+                actor = null; // FN
```

Comment:
I iterated over some solutions to fix this FN, however I gave up because it was adding needless complexity and it wasn't covering all cases

the root cause of this situation is the fact that:
- for each `finally` clause, to separate blocks are generated (see comments in `DeadStore` class)
- when analyzing a block that follow a `try` statement, we cannot tell whether it's a `catch`, a `finally` or just the code which follows the `try-catch-finally` statement

Reply:
It might be worth capturing why this FN occurs in the test.

Reply:
fair point, I'll add these comments in the test

Reply:
As discussed this FN can be fixed by adding finally clause in the instructions of the finally block to distinguish catch from finally.